scripts/dev/unscrew-werror/clang

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+#
+# C/C++ and `-Werror` are the worst. "Let's have our software stop compiling,
+# every time some toolchain somewhere decides to change some minor warning. Ugh..."
+
+set -eouo pipefail
+
+if [ -z "${UNSCREW_WERROR_ORIG:-}" ]; then
+	>&2 echo "Must set UNSCREW_WERROR_ORIG" to the original clang binary
+	exit 1
+fi
+
+args=()
+for arg in "$@"; do
+    [[ "$arg" != "-Werror" ]] && args+=("$arg")
```

Comment:
Otherwise `honggfuzz` crate can't compile with newer toolchains:

```
  mangle.c:428:6: error: initializer-string for character array is too long, array size is 8 but initializer has size 9 (including the null terminating charact
er); did you mean to use the 'nonstring' attribute? [-Werror,-Wunterminated-string-initialization]
    428 |     {"\x00\x00\x00\x00\x00\x00\x00\x10", 8},
        |      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  fatal error: too many errors emitted, stopping now [-ferror-limit=]
```